openpype/hosts/blender/api/plugin.py

```diff
@@ -281,6 +543,152 @@ def _load(self,
 
         # return self._get_instance_collection(instance_name, nodes)
 
+    @staticmethod
+    def _get_last_representation(asset_group):
+        current_representation = legacy_io.find_one({
+            "_id": ObjectId(asset_group[AVALON_PROPERTY]["representation"])
+        })
+        current_version, subset, asset, project = legacy_io.parenthood(
+            current_representation
+        )
+        last_version = legacy_io.find_one(
+            {"type": "version", "parent": subset["_id"]},
+            sort=[("name", -1)]
+        )
+        last_representation = legacy_io.find_one({
+            "type": "representation",
+            "parent": last_version["_id"],
+            "name": current_representation["name"]
+        })
+
+        assert last_representation is not None, "Representation wasn't found"
+
+        return last_representation
+
+    @classmethod
+    def is_updated(cls, asset_group, libpath=None):
+        """Check data before update. Return True if already updated"""
+
+        if libpath is None:
+            last_representation = cls._get_last_representation(asset_group)
+            libpath = get_representation_path(last_representation)
+
+            assert libpath, (
+                f"No existing library file found for {asset_group.name}"
+            )
+
+        assert Path(libpath).is_file(), (
+            f"The library file doesn't exist: {libpath}"
+        )
+        assert Path(libpath).suffix.lower() in VALID_EXTENSIONS, (
+            f"Unsupported library file: {libpath}"
+        )
+
+        group_libpath = asset_group[AVALON_PROPERTY]["libpath"]
+
+        normalized_group_libpath = (
+            str(Path(bpy.path.abspath(group_libpath)).resolve())
+        )
+        normalized_libpath = (
+            str(Path(bpy.path.abspath(libpath)).resolve())
+        )
+        cls.log.debug(
+            f"normalized_group_libpath:\n  {normalized_group_libpath}\n"
+            f"normalized_libpath:\n  {normalized_libpath}"
+        )
+        return normalized_group_libpath == normalized_libpath
+
+    def _update_blend(self, container: Dict, representation: Dict):
+        """Update the loaded asset.
+
+        This will remove all objects of the current collection, load the new
+        ones and add them to the collection.
+        If the objects of the collection are used in another collection they
+        will not be removed, only unlinked. Normally this should not be the
+        case though.
+
+        Warning:
+            No nested collections are supported at the moment!
+        """
+        object_name = container["objectName"]
+        asset_group = (
+            bpy.data.collections.get(object_name) or
+            bpy.data.objects.get(object_name)
```

Comment:
Is it now possible an object will be used as an asset_group?

Reply:
Only with model representation loaded in layout or setdress (actually it is a very early behavior, we need to discuss this feat)

Reply:
With the current system, you containerize always with a `Collection` no? Then I don't see how the case you describe may happen.